### PR TITLE
Moved pod and operations from Controller to Kubelet

### PR DESCRIPTION
**Pull request by baghbidi**

### CODE REVIEW

**Review by Pixcell — APPROVED** [review]
LGTM

File: cmd/root.go [hunk 1]
```diff
@@ -58,7 +65,8 @@ const (
 var deleteNodeLock sync.Mutex
```
Comment (sergerad) [inline cmd/root.go]: var (
...
)

File: cmd/root.go [hunk 2]
```diff
@@ -92,7 +102,14 @@ This allows users to schedule kubernetes workloads on nodes that aren't running
...
 
-		ioFogKubelets = make(map[string]*controllertypes.IOFogKubelet)
+		ioFogKubelets = make(map[string]*IOFogKubelet)
+
+		controller = apps.IofogController{
+			Token:    controllerToken,
+			Endpoint: controllerUrl,
+		}
+
+		controllerClient, _ = client.NewWithToken(controllerUrl, controllerToken)
```
Comment (sergerad) [inline cmd/root.go]: why ignore error?

File: cmd/root.go [hunk 3]
```diff
@@ -163,15 +180,19 @@ func startKubelet(nodeId string) {
 		log.L.WithError(err).WithField("value", daemonPortEnv).Fatal("Invalid value from KUBELET_PORT in environment")
 	}
 
+	configMap := k8sClient.CoreV1().ConfigMaps(kubeNamespace)
+	store, err := api.NewKeyValueStore(configMap, configMapName)
```
Comment (sergerad) [inline cmd/root.go:187]: ignoring error here?

File: providers/iofog/broker.go [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+
+// DeletePod accepts a Pod definition and forwards the call to the iofog endpoint
+func (p *BrokerProvider) DeletePod(ctx context.Context, pod *v1.Pod) error {
+	if flowPod, err := p.getFlowPod(pod.Name); err != nil {
+		return err
+	} else {
+		if err = p.client.DeleteFlow(flowPod.FlowInfo.ID); err != nil {
+			return err
+		}
+		_ = p.store.Remove(pod.Name)
```
Comment (sergerad) [inline providers/iofog/broker.go]: What is the ignored return value? Can add comment explaining

File: providers/iofog/broker.go [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+// GetPodStatus retrieves the status of a given pod by name.
+func (p *BrokerProvider) GetPodStatus(ctx context.Context, namespace, name string) (*v1.PodStatus, error) {
+	flowPod, err := p.getFlowPod(name)
+	if err != nil {
+		return nil, err
+	}
+
+	microservices, err := p.client.GetMicroservicesPerFlow(flowPod.FlowInfo.ID)
+	if err != nil {
+		return nil, nil
```
Comment (sergerad) [inline providers/iofog/broker.go]: returning nil rather than err

**Review by sergerad — APPROVED** [review]

**Pull request merged**